### PR TITLE
added contact dialog

### PR DESCRIPTION
**Pull request by meomancer**
Added contact dialog to save/edit it to database

add:
giphy

Edit/delete:
giphy 1

### CODE REVIEW

File: cadasta/common/setting.py [hunk 1]
```diff
@@ -149,3 +149,26 @@ def get_path_data(organization_slug=None, project_slug=None):
             '%s.geojson' % project_slug
         )
     return data_path
+
+
+def get_path_database(database=None):
+    """ Path databased based on database.
```
Comment (ismailsunni) [inline cadasta/common/setting.py]: No space in the begining of the docstring. See : https://github.com/inasafe/inasafe/wiki/Coding-Standards#doc-strings-and-comments

File: cadasta/database/__init__.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+from cadasta.common.custom_logging import setup_logger
```
Comment (ismailsunni) [inline cadasta/database/__init__.py]: Add header file.

File: cadasta/database/cadasta_database.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+.. note:: This program is free software; you can redistribute it and/or modify
+     it under the terms of the GNU General Public License as published by
+     the Free Software Foundation; either version 2 of the License, or
+     (at your option) any later version.
+
+
+"""
+
+__author__ = 'Irwan Fathurrahman <irwan@kartoza.com>'
```
Comment (ismailsunni) [inline cadasta/database/cadasta_database.py]: Do not put code before import statement

File: cadasta/database/cadasta_database.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+from cadasta.common.setting import get_path_database
+
+LOGGER = logging.getLogger('CadastaQGISPlugin')
+
+database = 'QSQLITE'
+database_name = get_path_database('cadasta.db')
+
+
+class CadastaDatabase(object):
+    """Cadasta class to handle databasel."""
```
Comment (ismailsunni) [inline cadasta/database/cadasta_database.py]: Typo `databasel` ?

File: cadasta/database/cadasta_database.py [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+
+    def __init__(self):
+        """Constructor for the class."""
+
+    @staticmethod
+    def open_database():
+        """open Database
```
Comment (ismailsunni) [inline cadasta/database/cadasta_database.py]: Capital letter for first letter.

File: cadasta/database/cadasta_database.py [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+        if db:
+            db.setDatabaseName(database_name)
+            if db.open():
+                LOGGER.debug("not opened")
+                return db
+        return None
+
+    @staticmethod
+    def save_to_database(table, data):
+        """save Into database.
```
Comment (ismailsunni) [inline cadasta/database/cadasta_database.py]: Sentence case

File: cadasta/database/cadasta_database.py [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+                'TABLE': table, 'ID': ','.join(row_ids)
+            }
+        )
+        query = QtSql.QSqlQuery(db)
+        query.exec_(query_string)
+        db.close()
+
+    @staticmethod
+    def get_table_model(table):
+        """get table model of table.
```
Comment (ismailsunni) [inline cadasta/database/cadasta_database.py]: Use sentence case. There are some other also.

File: cadasta/gui/tools/widget/widget_base.py [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+__copyright__ = "Copyright 2016, Cadasta"
+__license__ = "GPL version 3"
+__email__ = "info@kartoza.org"
+__revision__ = '$Format:%H$'
+
+LOGGER = logging.getLogger('CadastaQGISPlugin')
+
+
+def get_widget_step_ui_class(py_file_name):
```
Comment (ismailsunni) [inline cadasta/gui/tools/widget/widget_base.py:30]: Docstring

File: cadasta/model/base_model.py [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+Cadasta project - **Base Model.**
+
+.. note:: This program is free software; you can redistribute it and/or modify
+     it under the terms of the GNU General Public License as published by
+     the Free Software Foundation; either version 2 of the License, or
+     (at your option) any later version.
+
+"""
+
+__author__ = 'Irwan Fathurrahman <irwan@kartoza.com>'
```
Comment (ismailsunni) [inline cadasta/model/base_model.py]: Import statement first

**Review by dimasciput — APPROVED** [review]
LGTM

**Pull request merged**